### PR TITLE
feat: Updating intro with goals and connections to other libraries

### PR DESCRIPTION
**Pull request by robertness**
Fixes #

Changes proposed in this pull request:

-

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/dodiscover/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/dodiscover/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,8 +42,29 @@ Further, DoDiscover seeks to help the user feel confident with their assumptions
...
 
-Causal representation learning is the task of learning high-level latent variables and the causal structure between them from low-level variables observed in data.
-DoDiscover seeks to support causal representation learning algorithms in the context of traditional causal discovery settings.
+DoDiscover is a Python library for causal discovery (causal structure learning).
+Our goal is to provide developers and researchers with guide rails for causal discovery that doesn't require deep knowledge of individual causal discovery algorithms.
+
+## What is the difference between dodiscover and causal-learn?
+
+The goal of dodiscover is to flatten the on-ramp to causal discovery algorithms.
+DoWhy provides a consistent API for various method for causal effect estimation.
```
Comment (bloebp) [inline README.md:51]: DoWhy goes beyond effect estimation with the structural causal models in the GCM module. What about "DoWhy provides a consistent API for various causal tasks, which typically require a graph structure."

File: README.md [hunk 2]
```diff
@@ -42,8 +42,29 @@ Further, DoDiscover seeks to help the user feel confident with their assumptions
...
+DoDiscover aims to do the same with causal discovery algorithms.
```
Comment (bloebp) [inline README.md:52]: Accordingly: "DoDiscover aims to provide a cohesive and user-friendly API to apply causal discovery algorithms for inferring a causal graph from data."

File: README.md [hunk 3]
```diff
@@ -42,8 +42,29 @@ Further, DoDiscover seeks to help the user feel confident with their assumptions
...
+In constrast, dodiscover's focus is on the discovery API and usability.
+
+When possible, dodiscover prefers to provide an API wrapper to discovery algorithms in causal-learn and other libraries.
+Please consider contributing to [causal-learn](https://github.com/py-why/causal-learn) if you plan to implement an algorithm from scratch, then contributing a wrapper in dodiscover.
+
+## What is the relationship with pywhy-graphs and py-indep?
+
+[pywhy-graphs](https://github.com/py-why/pywhy-graphs) is the home of graph data structures and graph algorithms in PyWhy.
+
+[py-indep](https://github.com/py-why/py-indep) is the home of implementations of tests for conditional independence and other constraints used in tasks including but not limited to causal discovery.
```
Comment (bloebp) [inline README.md:67]: What about:
"... serves as a repository for implementations of (un)conditional independence tests, which can be utilized in various tasks, such as causal discovery."

File: README.md [hunk 4]
```diff
@@ -42,8 +42,29 @@ Further, DoDiscover seeks to help the user feel confident with their assumptions
 
 ## Unite causal discovery and causal representation learning
 
-Causal representation learning is the task of learning high-level latent variables and the causal structure between them from low-level variables observed in data.
-DoDiscover seeks to support causal representation learning algorithms in the context of traditional causal discovery settings.
+DoDiscover is a Python library for causal discovery (causal structure learning).
+Our goal is to provide developers and researchers with guide rails for causal discovery that doesn't require deep knowledge of individual causal discovery algorithms.
+
+## What is the difference between dodiscover and causal-learn?
```
Comment (adam2392) [inline README.md:48]:
```suggestion
## What is the difference between dodiscover and other pywhy packages?
```

**Review by adam2392 — APPROVED** [review]
LGTM after changes are made

File: README.md [hunk 5]
```diff
@@ -42,8 +42,29 @@ Further, DoDiscover seeks to help the user feel confident with their assumptions
...
+
+The goal of dodiscover is to flatten the on-ramp to causal discovery algorithms.
+DoWhy provides a consistent API for various method for causal effect estimation.
+DoDiscover aims to do the same with causal discovery algorithms.
+
+[causal-learn](https://github.com/py-why/causal-learn) is an extensive collection of causal discovery algorithms.
+It continuous to host new cutting-edge algorithms in causal discovery.
+However, these algorithms do not have a unified API.
+Further, the historic focus of causal-learn is increasing the capabilities of discovery algorithms.
+In constrast, dodiscover's focus is on the discovery API and usability.
```
Comment (adam2392) [inline README.md:58]: Perhaps we can mention: In the future we plan on trying to integrate the two libraries.

File: README.md (hunk 1 above)
Comment (robertness) [reply README.md]: [570e5c0](https://github.com/py-why/dodiscover/pull/132/commits/570e5c0f8b663dd28404a7ea771cd16f5997d1ce)

File: README.md (hunk 2 above)
Comment (robertness) [reply README.md]: [db3001a](https://github.com/py-why/dodiscover/pull/132/commits/db3001a882929281d1710856a90e539c8112e52d)

File: README.md (hunk 3 above)
Comment (robertness) [reply README.md]: [03e3f5a](https://github.com/py-why/dodiscover/pull/132/commits/03e3f5ad30c34df702c712b1ec91d13763b9a5a6)

File: README.md (hunk 5 above)
Comment (robertness) [reply README.md:58]: https://github.com/py-why/dodiscover/pull/132/commits/abf3bebeea37ac1e0ba0425d8be2d99c08a848d9

File: README.md [hunk 6]
```diff
@@ -42,8 +42,31 @@ Further, DoDiscover seeks to help the user feel confident with their assumptions
...
+
+The goal of dodiscover is to flatten the on-ramp to causal discovery algorithms.
+DoWhy provides a consistent API for various causal tasks that typically require a graph structure.
+Similarly, DoDiscover aims to provide a cohesive and user-friendly API to apply causal discovery algorithms for inferring a causal graph from data.
+
+[causal-learn](https://github.com/py-why/causal-learn) is an extensive collection of causal discovery algorithms.
+It continuous to host new cutting-edge algorithms in causal discovery.
+However, these algorithms do not have a unified API.
+Further, the historic focus of causal-learn is increasing the capabilities of discovery algorithms.
+In constrast, dodiscover's focus is on the discovery API and usability.
```
Comment (jaron-lee) [inline README.md:58]: Small typo: "contrast"

File: README.md [hunk 7]
```diff
@@ -42,8 +42,31 @@ Further, DoDiscover seeks to help the user feel confident with their assumptions
...
+
+When possible, dodiscover prefers to provide an API wrapper to discovery algorithms in causal-learn and other libraries.
+Please consider contributing to [causal-learn](https://github.com/py-why/causal-learn) if you plan to implement an algorithm from scratch, then contributing a wrapper in dodiscover.
+
+In the future we plan on trying to integrate the two libraries.
+
+## What is the relationship with pywhy-graphs and py-indep?
```
Comment (jaron-lee) [inline README.md:65]: I think the new library is to be called pywhy-stats, if I followed the discussion on discord

**Review by jaron-lee — APPROVED** [review]
Minor nits but LGTM

**Pull request merged**